### PR TITLE
Add timeline_id to detection rules

### LINKED ISSUE #92: [Rule Tuning] Network rules should use the new 'Generic Network Timeline'
Where appropriate, we want to support Timeline Templates (https://github.com/elastic/kibana/pull/67496) out of the box. This issue is to track network rules.

### PR DESCRIPTION
**Pull request by dcode**
## Issues
- [X] Add timeline to network rules (Closes #92)
- [x] Add timeline to endpoint rules (#93)
- [x] Add timeline to process rules (#94)

## Summary
Kibana 7.9 adds support for timeline templates (elastic/kibana#67496). This PR associates the default generic templates with rules where they make sense. This should make the analytic workflow smoother when analyzing detections that we ship out of the box.

## Contributor checklist

- [X] Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- [X] Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by dcode** [comment]
blocked on https://github.com/elastic/kibana/issues/74161

File: rules/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # rules/
 
-Rules within this folder are organized by solution or platform. The structure is flattened out, because nested file hierarchies are hard to navigate and find what you're looking for. Each directory contains several [.toml](https://github.com/toml-lang/toml) files, and the primary ATT&CK tactic is included in the file name when it's relevant (i.e. [`windows/execution_via_compiled_html_file.toml`](windows/execution_via_compiled_html_file.toml))  
+Rules within this folder are organized by solution or platform. The structure is flattened out, because nested file hierarchies are hard to navigate and find what you're looking for. Each directory contains several [.toml](https://github.com/toml-lang/toml) files, and the primary ATT&CK tactic is included in the file name when it's relevant (i.e. [`windows/execution_via_compiled_html_file.toml`](windows/execution_via_compiled_html_file.toml))
```
Comment (rw-access) [inline rules/README.md:3]: what changed on this line?

**Review by rw-access — APPROVED** [review]
cool beans

File: rules/README.md (hunk 1 above)
Comment (dcode) [reply rules/README.md:3]: trailing space was deleted. I'm happy to put it back

Comment (dcode) [reply rules/README.md:3]: during difficulties of the line ending stuff, I had done a blanket "put back" of extra trailing newlines, and the editor deleted the trailing spaces on each line.

Comment (rw-access) [reply rules/README.md:3]: ah no need to put it back. i was looking at this earlier on my phone and its highlighting isn't smart enough to show whitespace

**Comment by bm11100** [comment]
The `templateTimelineId` matches what's in Kibana. However, when looking at the templates, they have a specified `start` and `end` date. Each time a Generic timeline is opened, it defaults to the hardcoded date range (all are in April 2020). This isn't ideal and may be a blocker until adjusted. 

image

image

**Comment by dcode** [comment]
We're actually good on this now. I met with @XavierM this morning and discussed. When a user clicks to "investigate in timeline" on an alert, the datetime range is updated to fit the current time window being investigated. I'll re-merge and push back to the branch to bring it up to date.

**Review by bm11100 — APPROVED** [review]
LGTM

**Pull request merged**